network/netcore/src/transport/tcp.rs

```diff
@@ -74,18 +74,21 @@ impl Transport for TcpTransport {
         &self,
         addr: NetworkAddress,
     ) -> Result<(Self::Listener, NetworkAddress), Self::Error> {
-        let socket_addr = multiaddr_to_socketaddr(&addr)?;
+        let (socket_addr, addr_suffix) = parse_socketaddr(&addr)?;
         let config = self.clone();
+
         let listener = ::std::net::TcpListener::bind(&socket_addr)?;
-        let local_addr = NetworkAddress::from(listener.local_addr()?);
         let listener = TcpListener::try_from(listener)?;
 
+        let mut actual_addr = NetworkAddress::from(listener.local_addr()?);
+        actual_addr.extend_from_slice(addr_suffix);
```

Comment:
maybe extend_from_slice could consume and return

  extend_from_slice(val: mut T) -> T